### PR TITLE
patch perk Ultimate Leader

### PR DESCRIPTION
**Pull request by iPherian**
fixes Tyler-IN/MnB2-Bannerlord-CommunityPatch#275

- Base game has duplicate perks, apply when either are active

- Remove requirement of being head of faction or independent

### CODE REVIEW

**Comment by Solganis** [comment]
I can test it asap, if someone approve it to the Dev.

File: src/CommunityPatch/Patches/Perks/Social/Leadership/UltimateLeaderPatch.cs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      return false;
+    }
+
+    private static PerkObject GetUltimateLeadershipPerk(IReadOnlyList<PerkObject> probablePerks) {
+      try {
+        // ReSharper disable once PossibleNullReferenceException
+        return (PerkObject) typeof(DefaultPerks.Leadership)
+          .GetProperty("UltimateLeader", Public | NonPublic | Static | DeclaredOnly)
+          .GetGetMethod()
+          .Invoke(null, null);
```
Comment (Tyler-IN) [inline src/CommunityPatch/Patches/Perks/Social/Leadership/UltimateLeaderPatch.cs:105]: DefaultPerks will be wrong after loading saved games, sometimes but _not always._
Have thoroughly investigated problems with it, doesn't apply to Feats or Traits or Skills, just Perks.
Recently had to fix a case of it seemingly mysteriously breaking after 2nd load for horse bow / crossbow perk patches.
Screenshot of original discovery pasted in https://github.com/Tyler-IN/MnB2-Bannerlord-CommunityPatch/pull/182#issuecomment-616277470

File: src/CommunityPatch/Patches/Perks/Social/Leadership/UltimateLeaderPatch.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.SandBox.GameComponents.Party;
+using TaleWorlds.Core;
+using TaleWorlds.Localization;
+using static System.Reflection.BindingFlags;
+using static CommunityPatch.PatchApplicabilityHelper;
+
+namespace CommunityPatch.Patches.Perks.Social.Leadership {
+
+  public sealed class UltimateLeaderPatch : PatchBase<UltimateLeaderPatch> {
```
Comment (Tyler-IN) [inline src/CommunityPatch/Patches/Perks/Social/Leadership/UltimateLeaderPatch.cs:15]: There's also `PerkPatchBase<>` you could use now.
Might need to be updated to handle the scenario of duplicate perk IDs, but it _might_ work ok.

File: src/CommunityPatch/Patches/Perks/Social/Leadership/UltimateLeaderPatch.cs (hunk 1 above)
Comment (iPherian) [reply src/CommunityPatch/Patches/Perks/Social/Leadership/UltimateLeaderPatch.cs]: Good to know. Added a change.

**Pull request merged**